### PR TITLE
wrapper for gdalmdimtranslate

### PR DESCRIPTION
**Pull request by ChristianBeilschmidt**
- [X] I agree to follow the project's [code of conduct](https://github.com/georust/gdal/blob/master/CODE_OF_CONDUCT.md).
- [X] I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.
---

I created a wrapper around `GDALMultiDimTranslate`. I used part of the logic for the wrapper for the VRT program.

### CODE REVIEW

File: src/programs/raster/mdimtranslate.rs [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+        unsafe {
+            gdal_sys::GDALMultiDimTranslateOptionsFree(self.c_options);
+        }
+    }
+}
+
+impl TryFrom<Vec<&str>> for MultiDimTranslateOption {
+    type Error = GdalError;
+
+    fn try_from(value: Vec<&str>) -> std::result::Result<Self, Self::Error> {
```
Comment (lnicola) [inline src/programs/raster/mdimtranslate.rs:79]: Does this work?

```suggestion
    fn try_from(value: Vec<&str>) -> Result<Self> {
```

File: src/programs/raster/mdimtranslate.rs [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+    mem::ManuallyDrop,
+    path::Path,
+    ptr::{null, null_mut},
+};
+
+/// Wraps a [GDALMultiDimTranslateOptions] object.
+///
+/// [GDALMultiDimTranslateOptions]: https://gdal.org/api/gdal_utils.html#_CPPv428GDALMultiDimTranslateOptions
+///
+pub struct MultiDimTranslateOption {
```
Comment (lnicola) [inline src/programs/raster/mdimtranslate.rs:21]: Should probably be called `MultiDimTranslateOptions`, since there can be multiple options inside.

Comment (ChristianBeilschmidt) [reply src/programs/raster/mdimtranslate.rs]: [23957df](https://github.com/georust/gdal/pull/289/commits/23957df6195fc5557422ffd81aa724c7aa4aaaf1)

File: src/programs/raster/mdimtranslate.rs (hunk 1 above)
Comment (ChristianBeilschmidt) [reply src/programs/raster/mdimtranslate.rs]: [23957df](https://github.com/georust/gdal/pull/289/commits/23957df6195fc5557422ffd81aa724c7aa4aaaf1)

**Review by lnicola — APPROVED** [review]

**Comment by ChristianBeilschmidt** [comment]
I forgot to replace the `Cell<bool>` with a plain boolean, I guess it can be this simple :see_no_evil: .

**Comment by lnicola** [comment]
```
thread 'config::tests::test_error_handler' panicked at 'assertion failed: `(left == right)`
  left: `[(Warning, 1, "GPKG: bad application_id=0x00000000 on '/tmp/.tmpPTiAzK.gpkg'"), (Failure, 42, "foo"), (Warning, 1, "bar")]`,
 right: `[(Failure, 42, "foo"), (Warning, 1, "bar")]`', src/config.rs:230:9
```

Is this some sort of new GDAL warning?

**Comment by ChristianBeilschmidt** [comment]
> ```
> thread 'config::tests::test_error_handler' panicked at 'assertion failed: `(left == right)`
>   left: `[(Warning, 1, "GPKG: bad application_id=0x00000000 on '/tmp/.tmpPTiAzK.gpkg'"), (Failure, 42, "foo"), (Warning, 1, "bar")]`,
>  right: `[(Failure, 42, "foo"), (Warning, 1, "bar")]`', src/config.rs:230:9
> ```
> 
> Is this some sort of new GDAL warning?

Since it appeared in the LTS-CI, I guess not.

What could have happened:
The tests run in parallel. Some test sets the global error handler. Then some other test fails… which lands in the error handler. And this test gets some strange results.

I'm not sure, but we could get races on all the global stuff.
Maybe we need some global Mutexes on some test stuff that must not interfere with something else :shrug:.

What do you think?

**Comment by jdroenner** [comment]
bors merge

**Comment by lnicola** [comment]
I'm a bit worried about the test failure here.

**Comment by jdroenner** [comment]
i guess the problem is not really related to this PR but to gdal and cargo test in general. Cargo runs all the tests in parallel and when two tests fail at the same time we have a problem with global state.

**Comment by lnicola** [comment]
I think we can move them to an integration test.

**Pull request merged**